__tests__/integration/hathorwallet_others.test.ts

```diff
@@ -18,9 +19,88 @@ import dateFormatter from '../../src/utils/date';
 import { AddressError } from '../../src/errors';
 import { precalculationHelpers } from './helpers/wallet-precalculation.helper';
 import { ConnectionState } from '../../src/wallet/types';
+import HathorWallet from '../../src/new/wallet';
 
 const fakeTokenUid = '008a19f84f2ae284f19bf3d03386c878ddd15b8b0b604a3a3539aa9d714686e1';
 
+describe('processing transaction metadata changes', () => {
+  let hWallet: HathorWallet;
+  let walletData: { words: string; addresses: string[] };
+
+  beforeAll(async () => {
+    walletData = precalculationHelpers.test.getPrecalculatedWallet();
+  });
+
+  beforeEach(async () => {
+    hWallet = await generateWalletHelper({
+      seed: walletData.words,
+      addresses: walletData.addresses,
+    });
+  });
+
+  afterEach(async () => {
+    await hWallet.stop();
+  });
+
+  it('should process entire history and balance when a tx is voided', async () => {
+    const addr0 = await hWallet.getAddressAtIndex(0);
+    const wsSpy: jest.SpiedFunction<typeof hWallet.onNewTx> = jest.spyOn(hWallet, 'onNewTx');
+    const procSpy: jest.SpiedFunction<typeof hWallet.storage.processHistory> = jest.spyOn(
+      hWallet.storage,
+      'processHistory'
+    );
+
+    await expect(hWallet.getBalance(NATIVE_TOKEN_UID)).resolves.toEqual([
+      expect.objectContaining({
+        token: expect.objectContaining({ id: NATIVE_TOKEN_UID }),
+        transactions: 0,
+        balance: { unlocked: 0n, locked: 0n },
+      }),
+    ]);
+
+    const injectedTx = await GenesisWalletHelper.injectFunds(hWallet, addr0, 10n);
+    if (!injectedTx.hash) {
+      throw new Error('Could not inject funds into wallet');
+    }
+    await waitForTxReceived(hWallet, injectedTx.hash);
+
+    await expect(hWallet.getBalance(NATIVE_TOKEN_UID)).resolves.toEqual([
+      expect.objectContaining({
+        token: expect.objectContaining({ id: NATIVE_TOKEN_UID }),
+        transactions: 1,
+        balance: { unlocked: 10n, locked: 0n },
+      }),
+    ]);
+
+    expect(wsSpy).toHaveBeenCalled();
+
+    const { lastCall } = wsSpy.mock;
+    expect(lastCall).toBeDefined();
+    if (!lastCall) {
+      throw new Error('Unexpected error');
+    }
```

Comment:
suggestion(non-blocking): Lines 79 to 81 are redundant: if `!lastCall`, jest will have already finished the test with a failure.

Reply:
Yes, but typescript type system does not know this.
The `expect` statement was to make sure the test fails gracefully and the if statement was to ensure the typing.